src/components/target/index.js

```diff
@@ -0,0 +1,22 @@
+import { useState } from 'react';
+
+import { Marker, useMapEvents } from 'react-leaflet';
+import myIcon from '../Map/Icon';
+
+const AddMarkerToClick = props => {
+  const map = useMapEvents({
+    click(e) {
+      props.targets.push(e.latlng);
```

Comment:
Remember not to mutate the `targets` state directly, you should update it using `setTargets`

Though I'm not sure if you should be adding the coordinates of the new target to the targets array when you click on the map.